‘DOC-2652-gsql-files-output-4.2’ (#335)
tigergraph/gsql-docs: modules/querying/pages/declaration-and-assignment-statements.adoc

[#_file_objects]
=== `FILE` objects

> A `FILE` object is a sequential text storage object, associated with a text file on the local machine or with an S3 bucket.

Reviewer: ```suggestion
A `FILE` object is a sequential text storage object, associated with a text file on the local machine or with a cloud storage bucket.
```
S3 is for AWS only. Microsoft and Google have similar cloud object storage services that we will likely support in the future.

include::appendix:example$work_net/declaration_file_object_query.gsql[]
----

=== S3 object
==== Define s3 file object

> The path should start with `s3://`, followed by the bucket name and the S3 path, e.g., `s3://bucket-name/queryoutput/output.csv`. During the execution of the query, content will be uploaded to the S3 bucket. Note that the S3 object cannot be modified or appended, if an S3 object with the same path already exists, it will be overwritten.

Reviewer: ```suggestion
The path should start with `s3://`, followed by the bucket name and the S3 path, e.g., `s3://bucket-name/queryoutput/output.csv`. During the execution of the query, content will be uploaded to the S3 bucket. Append mode writing is not supported.
If an S3 object with the same path already exists, it will be overwritten.

```
